### PR TITLE
feat: add Vertex.ai generative AI tasks

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: src/main/java/io/kestra/plugin/gcp/vertexai/ChatCompletion.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        @Example(
+            title = "Chat completion using the Vertex AI PaLM API",
+            code = {
+                """
+                    region: us-central1
+                    projectId: my-project
+                    context: I love jokes that talk about sport
+                    messages:
+                      - author: user
+                        content: Please tell me a joke"""
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/gcp/vertexai/ChatCompletion.java:46]:
```suggestion
                        content: tell me a joke"""
```

you're so polite 😄

**Review by anna-geller — APPROVED** [review]
Awesome!

**Comment by anna-geller** [comment]
I only wonder whether we would need to expose `serviceAccount` property? or does it happen automatically? all other GCP tasks and triggers have that

File: src/main/java/io/kestra/plugin/gcp/vertexai/ChatCompletion.java (hunk 1 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/gcp/vertexai/ChatCompletion.java:46]: Well, it's a habit I took always to be polite with bots, you never know ... Skynet...

**Comment by loicmathieu** [comment]
@anna-geller 

> I only wonder whether we would need to expose serviceAccount property? or does it happen automatically? all other GCP tasks and triggers have that

It is exposed but not included in the examples as they are not full examples.

**Review by Skraye — APPROVED** [review]
Miss some tests
however, LGTM

**Pull request merged**